### DISCUSSION TITLE
Using local assets as cache

### DISCUSSION THREAD

**Discussion by shalak** [antonym/netboot.xyz#1397] [Ideas]
I think it would be great, to have the following workflow possible:
- on the first boot of e.g. ubuntu 22.04, its installer is downloaded from upstream, online source
- the download happens simultaneously to both the device being booted over PXE as well as the assets folder of netboot.xyz container
- any subsequent PXE boot will use the assets from local network, saving both bandwidth and time 

From what I understand, this is currently impossible, and the only way is to either use the upstream, or local asset source, which needs to be populated before-hand.

**Comment by michaelblasius** [comment]
could local and remote assets not be completely separate?
- when an image is selected to boot into, the system would check the local assets before pulling from the web
- if an image needs to get pulled from the web, it gets put into the local assets then sent from local to the device
- the next time that image is selected it would be run from the local copy right away

**Reply by shalak** [reply]
That's pretty much what I meant :)

**Reply by MaKaNu** [reply]
Hey we also had this issue and saw the possiblity for caching. What does @antonym think about this?

**Comment by saltedlolly** [comment]
This feature seems like such an obvious addition. To be honest I was amazed it didn't already do this when i first installed it. Has anyone found a workaround?

**Comment by saltedlolly** [comment]
Issue #1220 discusses some workarounds for this. It would be nice if this could be setup automatically.
